backend/src/main/java/be/cytomine/appstore/services/TaskService.java

```diff
@@ -105,9 +105,17 @@ public Optional<TaskDescription> uploadTask(MultipartFile taskArchive)
                 new StorageData(uploadTaskArchive.getDescriptorFile(), "descriptor.yml")
             );
             log.info("UploadTask: descriptor.yml is stored in object storage");
+
+            if (Objects.nonNull(uploadTaskArchive.getLogo())) {
+                fileStorageHandler.saveStorageData(
+                    storage,
+                    new StorageData(uploadTaskArchive.getLogo(), "logo.png")
+                );
+                log.info("UploadTask: logo.png is stored in object storage");
+            }
         } catch (FileStorageException e) {
             try {
-                log.info("UploadTask: failed to store descriptor.yml");
+                log.info("UploadTask: failed to store descriptor.yml or logo.png in object storage");
```

Comment:
I would suggest to add the error message in addition.
In case of failure, we do not know whether it fails to store the descriptor or the logo